core/src/main/java/org/apache/carbondata/core/metadata/schema/table/DiskBasedDMSchemaStorageProvider.java

```diff
@@ -97,7 +97,7 @@ public void saveSchema(DataMapSchema dataMapSchema) throws IOException {
   @Override
   public DataMapSchema retrieveSchema(String dataMapName)
       throws IOException, NoSuchDataMapException {
-    checkAndReloadDataMapSchemas(true);
+    checkAndReloadDataMapSchemas(false);
```

Comment:
If there is no mdtFile present, then if we pass `touchFile` as false, then everytime `datamapSchemas` cache will be loaded from disk. I think it may impact the performance